README.md

```diff
@@ -0,0 +1,177 @@
+# Programming Arcana
+
+The Programming Arcana is an open source programming text book that focuses on teaching introductory programming concepts.
+
+The guiding principles are:
+
+- Transferable concepts are more important than language specifics
+- Build concepts on top of known concepts as much as possible
+- Focus on procedural programming initially, with a view to transitioning to objects
+- Build programs that make it easy to see what is happening in the code
+
+# Code formatter
+
+We use [_clang-format_](https://clang.llvm.org/docs/ClangFormat.html) as our code format for this project.
+
+## Installation
+
+### Linux
+
+_clang-format_ can be installed by:
+
+```bash
+sudo apt install clang-format
+```
+
+### macOS
+
+_clang-format_ can be installed by:
+
+```bash
+brew install clang-format
+```
+
+### Windows
+
+The executable file can be downloaded from [llvm org](https://llvm.org/builds/).
+
+### Other
+
+The _clang-format_ can also be installed with python and nodejs:
+
+```bash
+# python
+python3 -m pip install clang-format
+
+# nodejs
+npm install -g clang-format
+```
+
+The [VS code extension](https://marketplace.visualstudio.com/items?itemName=xaver.clang-format). (optional, clang-format required)
+
+## The format style
+
+The base format style used is [_google_](https://google.github.io/styleguide/cppguide.html).
+
+## Usage
+
+To use the code formatter you can use command:
+
+```bash
+clang-format -i <file name>
+```
+
+Or if you have VS code extension installed you can use _right click_ -> _Format Document With ..._ option.
+
+In order to recursively format all files, you can use command:
+
+```bash
+make format
```

Comment:
**question:** Can the wrapper script format the code as well? I think we should either go with `make` wrapper for the clang-format command or the wrapper script (see more about my [thought](https://github.com/thoth-tech/programming-arcana/pull/2/files#r867419985) to stay with the earlier)

Reply:
sure thing

Reply:
Yes we can use wrapper script to format code.From my opinoin we can use the wrapper script and remove the `makefile`. Then intoduce the command ` find . -regex '.*\.\(cpp\|c\)' -exec clang-format -style=file -i {} \;` in readme for windows users as my [reply](https://github.com/thoth-tech/programming-arcana/pull/2#discussion_r867529693) because windows users need wsl to running the ` make ` command. So I think there's no point to have a ` makefile ` with 
```
format: 
	find . -regex '.*\.\(cpp\|c\)' -exec clang-format -style=file -i {} \;
```

Reply:
Is `find` available in Windows or do we still need WSL?

Reply:
Ops forget the ` find ` command is not available in windows, I will try to use powershell command instead.

And unfortunate clang-format does not support recurse function.